### DISCUSSION TITLE
my github.io page does not show up in google search

### DISCUSSION THREAD

**Discussion by MMorafah** [community/community#44421] [Other Feature Feedback, Questions, & Ideas] [answered]
### Github pages

Question

### Body

Hello, 

I have created a personal website using Jekyll. (Github repo: https://github.com/MMorafah/homepage) and (website: https://mmorafah.github.io/homepage/). 

I has passed more than 6 month and still have not showed up in google search. Could you please help? 

Thank you very much in advance.

**Comment by melvincwng** [comment, marked as answer]
Hey @MMorafah, welcome to GitHub Community 👋🏻 

To address your question, I did a quick google search on my end, and below are my findings:

![image](https://user-images.githubusercontent.com/77479885/212583775-086d45af-7d0c-4a5f-b8fb-3d3e902011c1.png)

First of all, your GitHub profile and GitHub personal website have already been indexed on Google Search, and is searchable by other users already.

Second of all, it takes time for Google Search engine's web crawlers to crawl through websites & index them in the search engine. This process takes time, and as quoted from this [documentation](https://developers.google.com/search/docs/crawling-indexing/ask-google-to-recrawl#:~:text=Crawling%20can%20take%20anywhere%20from,instantly%20or%20even%20at%20all.)  by Google, "it can take anywhere from a few days to weeks". Just to let you know, there isn't much the community members here can do to help expedite the process of indexing your website on Google. Just informing you in case you intend to publish new websites, you'll just have to wait patiently for your website to be indexed.

Hope this clarifies & answers your question 😄.

Thank you, and have a good day ahead 😃!

**Reply by MMorafah** [reply]
Hey @melvincwng, 

Thanks a lot for your comment. I would like google search show my website when I put my name "Mahdi Morafah" in the search. There are people who I search their name, show their github pages website. Could you please let me know if you have any information on this? 

Thank you very much in advance :)

**Reply by melvincwng** [reply]
@MMorafah, Google Search ranks their results based on various metrics such as relevancy & the search terms that were being used.

You might have to improve the SEO of your website if you wish for others to find your website through your name "Mahdi Morafah" as the search term. SEO is something that you can probably research and try to optimize for your GitHub webpage.

For SEO-related matters, if you need some detailed information/references, consider the following:
1. [Google's SEO documentation](https://developers.google.com/search/docs/fundamentals/seo-starter-guide?hl=en&ref_topic=4564315&visit_id=638094469727299827-3185838056&rd=2)
2. [StackOverFlow Question - How to get GitHub Pages website on Google Search](https://stackoverflow.com/questions/49073043/get-github-pages-site-found-in-google-search-results)

All the best, take care and good luck 👍🏻

**Reply by melvincwng** [reply]
Btw, I'm not very familiar with SEO. If it's SEO related, perhaps another community member who's an expert in SEO, can provide their inputs on this 😃

**Reply by novaTopFlex** [reply]
Also, I am seeing my GitHub profile on Bing Search but not on Google Search, and I believe it to be simply because Microsoft owns GitHub. However, I still see my GitHub profile name anyway.

**Reply by LiteBrite82** [reply]
The lighthouse report looks good, but there is an error in the console that looks like it could be security related; maybe an issue - I'm not familiar with it. 
![Screenshot 2024-08-08 at 11 18 16 AM](https://github.com/user-attachments/assets/d16d04ff-8d89-4bb5-b806-e3122d38e1cd)
![Screenshot 2024-08-08 at 11 21 01 AM](https://github.com/user-attachments/assets/ab85da92-cc6f-4448-971e-7ee168d4903c)

**Comment by kasir-barati** [comment]
@melvincwng I am not entirely convinced that's the case because I looked at his GitHub pages and I went to its "Blog" and saw tht OK there is a test blog post. So what I did was entering this in the `google.com/search?q=site:mmorafah.github.io+Hello world` and guess what, I got this "Your search - site:mmorafah.github.io Hello world - did not match any documents.". Now I am thinking that Google won't index pages like this: https://mmorafah.github.io/homepage/Blog/2022/distill/

Or is it because of how Jekyll/React works. I mean SPA and those kind of stories -- I am referring to SSR. Any idea?

**Notice: Maybe I am using google the wrong way and if I change my search query it will shows the result**

**Reply by taktsun** [reply]
Hi @MMorafah , I tried searching for your page as well, but like @kasir-barati , I can't search your blog page. It seems like https://mmorafah.github.io/homepage/ is the only page that so far showed up on google when I entered "site:mmorafah.github.io". But hey, now when I search your full name, the github page is there as well :) 

Have you figured out why this page was successfully on Google?

Btw, saying hi as (yet) another PhD student!

**Comment by weihan1** [comment]
Hey @MMorafah, I see that your Github pages pops up when I search your name, how did you fix that? I tried optimizing for keywords etc. but it still doesn't appear.

**Reply by tibordome** [reply]
Same issue here. Did you add some meta tags etc to get this fixed?

**Reply by David-Morales-Norato** [reply]
Hi @MMorafah, same as the other guys. Did you try something or did just suddenly it appear in Google Search?

**Reply by ErickNavarroD** [reply]
Hi! Was your problem solved by just waiting for google to index your pages?

**Reply by DinhKhai0201** [reply]
same for me. i changed my github username for 3 months ago. But google is still not indexed it

**Reply by weihan1** [reply]
Mine just got indexed like 4 months ago, it takes some time

**Comment by Franklyn-R-Silva** [comment]
If your GitHub Pages website is not showing up in Google search results, there are a few steps you can take to try to improve its visibility. Keep in mind that search engine indexing can take some time, so you might need to be patient as well. Here are some suggestions:

1. **Check Robots.txt**: Make sure that your website's `robots.txt` file is not blocking search engine crawlers from indexing your site. You can find this file at the root of your GitHub repository. It should not contain any rules that disallow search engine bots.

2. **Submit Sitemap**: Create a sitemap for your GitHub Pages site and submit it to Google Search Console. This helps Google understand the structure of your site and index it more effectively. GitHub Pages automatically generates a sitemap for your site, so you can find it at `https://<yourusername>.github.io/sitemap.xml`.

3. **Quality Content**: Ensure that your GitHub Pages site has quality and relevant content. Search engines prioritize sites that provide valuable and unique content to users.

4. **Keywords**: Use relevant keywords in your content that users might search for. This can help improve your site's visibility in search results.

5. **Backlinks**: Get links from other reputable websites to your GitHub Pages site. Backlinks from authoritative sources can improve your site's credibility and search engine ranking.

6. **Meta Tags**: Use appropriate meta tags, including title tags and meta descriptions, on your GitHub Pages site's HTML pages. These tags provide information to search engines about the content of your pages.

7. **Social Media Promotion**: Share your GitHub Pages site on social media platforms. Social signals can indirectly impact search engine rankings.

8. **Internal Linking**: Use internal linking within your site to guide visitors to other relevant pages on your GitHub Pages site. This can help search engines discover and index more of your content.

9. **Fetch as Google**: In Google Search Console, use the "Fetch as Google" tool to request that Googlebot crawl and index specific pages of your GitHub Pages site.

10. **Patience**: Sometimes it takes time for search engines to index new websites or changes to existing sites. Be patient and give the search engines time to discover and index your GitHub Pages site.

Remember that search engine optimization (SEO) is an ongoing process, and improving your site's visibility in search results may take time and effort. Additionally, Google's algorithms and indexing processes can change, so it's important to stay up-to-date with best practices for optimizing your site for search engines.

**Comment by nikolaydubina** [comment]
same here. when I used external domain for my github page go-cover-treemap.io it got indexed almost immediately, and got traffic from Google Search.

I removed custom domain, and it has been weeks now, and zero traffic from google search. 

* I allowed all robots.txt
* I monitored Google Analytics and Google Search Console

nothing seem to help. traffic to `<username>.github.io/<repo>` just goes straight to zero.

**Comment by nicoloceneda** [comment]
I have the same problem. I created my [website](https://nicoloceneda.github.io/) two months ago and it is still not visible, even if I search for the website content using quotation marks (e.g. "Welcome to the Black Swan, Nicolo Ceneda's webpage."). Is this normal?

**Reply by leydsonandrey** [reply]
You add your site on [https://search.google.com/u/4/search-console](https://search.google.com/u/4/search-console)?

**Reply by nicoloceneda** [reply]
I added it to the google search console, but it is still not visible.

**Reply by leydsonandrey** [reply]
> I added it to the google search console, but it is still not visible. 

Try adding a sitemap, and add it to search console.

Sitemap of my blog of exemple [https://leydsonandrey.github.io/sitemap.xml](https://leydsonandrey.github.io/sitemap.xml).

**Comment by GeoGC** [comment]


**Reply by leydsonandrey** [reply]
Create a `robots.txt` file on your static folder and put this:

For all search engines:

```
User-agent: *
Disallow: /
```

Only for Google Search:

```
User-agent: Googlebot
Disallow: /
```

**More info about `robots.txt`: https://developers.google.com/search/docs/crawling-indexing/robots/intro**

### Reference

- https://searchfacts.com/robots-txt-allow-disallow-all/
- https://developers.google.com/search/docs/crawling-indexing/overview-google-crawlers

**Reply by RDjarbeng** [reply]
Did this work for anyone?

**Comment by novaTopFlex** [comment]
If your GitHub.io pages do not appear in the upcoming future, then it may be that the world is most certainly losing the .io domain based on IANA rules.

**Comment by ElliotXie** [comment]
It has been two years, and my profile is still not searchable.

**Reply by yudhiesh** [reply]
Same here, funny enough my LinkedIn posts about my GitHub blogpost always shows up instead of my actual post.

**Reply by RDjarbeng** [reply]
You might have to open another discussion since this one has been marked as answered. I'm not sure it's actively being monitored

**Reply by martinvisintini** [reply]
I have same problem, google won't show my github post or profile, now I've created a google sites webpage with links to my github pages, we'll see if that works, in other hand today in a serendipia I tried to find my post using duckduck go, and voila: there were my github post. I will apreciate if some gan give some explanations here.
